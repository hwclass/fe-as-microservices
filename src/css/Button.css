.label { 
  color: green;
  font-size: 12px;
}
