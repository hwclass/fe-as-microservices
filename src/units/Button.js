import React from 'react'
import styled from 'styled-components';
import styles from '../css/Button.css'

const Button = ({ className }) => (
  <button className={className}>My litttle, tiny button</button>
)

const StyledButton = styled(Button)`
  color: palevioletred;
  font-weight: bold;
`;

export default StyledButton