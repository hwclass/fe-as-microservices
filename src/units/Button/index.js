import React from 'react'
import styled from 'styled-components';

const Button = ({ className }) => (
  <button className={className}>My litttle, tiny button</button>
)

const StyledButton = styled(Button)`
  color: green;
  font-size: 12px;
`;

export default StyledButton