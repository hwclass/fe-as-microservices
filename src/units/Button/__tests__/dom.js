// __tests__/dom.js

import React from 'react';
import styled from 'styled-components';
import 'jest-styled-components'
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Button from '../';

const StyledButton = styled(Button)``;

describe('Button unit', () => {
  test('renders without crashing', () => {
    shallow(<Button />);
  })

  test('parses the correct styled component', () => {
    const button = renderer.create(
      <Button>Send Message</Button>
    ).toJSON();
    expect(button).toMatchStyledComponentsSnapshot()
  })

  test('parses with the correct style rules', () => {
    const button = renderer.create(
      <Button>Send Message</Button>
    ).toJSON();
    expect(button).toHaveStyleRule('color', 'green')
    expect(button).toHaveStyleRule('font-size', '12px')
  })
})