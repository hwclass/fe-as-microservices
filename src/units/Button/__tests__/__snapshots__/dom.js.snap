// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button unit parses the correct styled component 1`] = `
.gsIPHz {
  color: green;
  font-size: 12px;
}

<button
  className="sc-bdVaJa gsIPHz"
>
  My litttle, tiny button
</button>
`;
